src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +140,101 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		string normalizedOrg = org.Replace( ".okta.com", "" );
```

Comment:
Similar amount of effort to how we do it in the [okta clients](https://github.com/Brightspace/bmx/blob/551f6cf7b1e0a3bdce7671706ceb102b848a5ed1/src/D2L.Bmx/Okta/OktaClient.cs#L55-L59)

Reply:
They are not equivalent.
Note: https://d2l.slack.com/archives/G4P099831/p1726600460156129?thread_ts=1726597940.542299&cid=G4P099831

Reply:
Ideally we can reuse the same logic in both places instead of writing it twice.

Reply:
Yea I didn't realize what you meant by that comment initially. I switched it